### PR TITLE
Avoid freeze removing blank lines from IMPROVED and END_INCLUSIVE UIA consoles

### LINKED ISSUE #14689: nvda logs a lot of '\r\n\r\n' if UIA support for Windows console is enabled
### Steps to reproduce:
1. In NVDA's Advanced Settings panel, make the following settings:
  - Windows Console support: combo box: UIA when available
2. Open cmd or git bash and press NVDA+Shift+M(Laptop);
### Actual behavior:
There are a lot of '\r\n' in the log.

### Expected behavior:
There won't be so many '\r\n' in the log
### NVDA logs, crash dumps and other attachments:
[log.txt](https://github.com/nvaccess/nvda/files/10866869/log.txt)

#### NVDA installed/portable/running from source:
Installed
#### NVDA version:
2023.1Beta2
#### Windows version:
Windows 10 22H2 (AMD64) build 19045.2604
#### Name and version of other software in use when reproducing the issue:
None
#### Other information about your system:
None
#### Does the issue still occur after restarting your computer?
Yes
#### Have you tried any other versions of NVDA? If so, please report their behaviors.
2022.4 is the same
#### If NVDA add-ons are disabled, is your problem still occurring?
Yes
#### Does the issue still occur after you run the COM Registration Fixing Tool in NVDA's tools menu?
Yes

### PR DESCRIPTION
**Pull request by ABuffEr**
### Link to issue number:

Fixes #14689

### Summary of the issue:

IMPROVED (apiLevel 1) and END_INCLUSIVE (apiLevel 0) UIA consoles produce a lot of blank lines (more than 17000) in some output, causing a NVDA freeze during speech dictionary processing.

### Description of user facing changes

None.

### Description of development approach

I added a _get_text method to ConsoleUIATextInfo class (for apiLevel 1 consoles), stripping all \r\n on the rightside of output from _get_text of superclass (implicitly used previously).
For apiLevel 0 consoles (ConsoleUIATextInfoWorkaroundEndInclusive class), I simply considered that the _get_text already present calls _get_text from superclass, that is, ConsoleUIATextInfo, so text is already cleaned.

### Testing strategy:

Manual, with #14689 str, in apiLevel 1 console.

### Known issues with pull request:

Hopefully none, this time.

### Change log entries:

Bug fixes
- When forcing UIA support with certain terminal and consoles, a bug is fixed which caused a freeze and the log file to be spammed.

### Code Review Checklist:

- [x] Pull Request description:
  - description is up to date
  - change log entries
- [x] Testing:
  - Unit tests
  - System (end to end) tests
  - Manual testing
- [x] API is compatible with existing add-ons.
- [ ] Documentation:
  - User Documentation
  - Developer / Technical Documentation
  - Context sensitive help for GUI changes
- [x] UX of all users considered:
  - Speech 
  - Braille
  - Low Vision
  - Different web browsers
  - Localization in other languages / culture than English
- [ ] Security precautions taken.

### CODE REVIEW

**Review by codeofdusk — APPROVED** [review]
I like this: it's a small, targeted fix.

That said, I thought constraining ourselves to the visible text worked around the microsoft/terminal#6986 issues? CC @carlos-zamora

**Comment by codeofdusk** [comment]
Also, consider a situation in which the console has (actual) empty lines at the end of output. If you call `rstrip`, it is impossible to distinguish between these actual empty text lines and the blank cells in the buffer.

**Comment by ABuffEr** [comment]
> Also, consider a situation in which the console has (actual) empty lines at the end of output. If you call `rstrip`, it is impossible to distinguish between these actual empty text lines and the blank cells in the buffer.

Yes, I considered it. But, for reasons I don't know exactly, even with no prompt string (that's beyond a condition of reasonable tradeoff, imho) I see blank lines of last output (I mean, I find them exploring with object navigation). I think exploration and voice work with slightly different sources.
However, I tested in apiLevel 0 console, not 1; I have Windows terminal too, but that falls under UIATextInfo without patches.

File: source/NVDAObjects/UIA/winConsoleUIA.py [hunk 1]
```diff
@@ -105,6 +105,12 @@ def _move(self, unit, direction, endPoint=None):
 		"Perform a move without respect to bounding."
 		return super(ConsoleUIATextInfo, self).move(unit, direction, endPoint)
 
+	def _get_text(self):
```
Comment (seanbudd) [inline source/NVDAObjects/UIA/winConsoleUIA.py:108]: Add typing

```suggestion
	def _get_text(self) -> str:
```

File: source/NVDAObjects/UIA/winConsoleUIA.py [hunk 2]
```diff
@@ -105,6 +105,12 @@ def _move(self, unit, direction, endPoint=None):
...
+		# #14689: IMPROVED and END_INCLUSIVE UIA consoles have many blank lines,
+		# which slows speech dictionary processing to a halt
+		res = super()._get_text()
+		return res.rstrip("\r\n")
```
[the remark this answers is not in the document]
Comment (ABuffEr) [reply source/NVDAObjects/UIA/winConsoleUIA.py:112]: Performance, and the fact it's a more targeted solution (even into striping only \r\n). I noticed that _get_text is called more times even when you write a single character in apiLevel 0 terminal (with no voice output). However, I can introduce it, if you think it's more reasonable.

[the remark this answers is not in the document]
Comment (seanbudd) [reply source/NVDAObjects/UIA/winConsoleUIA.py:112]: I am surprised there is a performance hit for checking the length of the tail, stripping is surely the most expensive operation.
My concern with this approach is that intentionally empty buffers that are just a few newlines might not be reported correctly.
For reference, you can probably cache `_get_text`, however that might cause other performance issues (not always getting the very latest text for a given core cycle).

[the remark this answers is not in the document]
Comment (ABuffEr) [reply source/NVDAObjects/UIA/winConsoleUIA.py:112]: Ok, I added previous check with IGNORE_TRAILING_WHITESPACE_LENGTH.

**Review by seanbudd — APPROVED** [review]

**Comment by seanbudd** [comment]
Thanks @ABuffEr

**Pull request merged**